sasmodels/generate.py

```diff
@@ -894,11 +894,17 @@ def contains_shell_volume(source):
             return True
     return False
 
+def _clean_source_filename(path):
+    # type: (str) -> str
+    base = dirname(__file__)
+    return path.replace(base, '.').replace('\\', '/')
```

Comment:
Maybe: `return "./"+basename(__file__)`

Otherwise `config/config.py` turns into `./..py`

Reply:
Yes indeed! I think I was trying to stay far too close to the original.

Reply:
Ah, I assumed `relative_to` would append leading `..` to get to a common base.

Since I find try/except blocks ugly I lean toward the string replacement interface, but then you should probably do:
```python
base, path = base.replace('\\', '/'), path.replace('\\', '/')
return path.replace(base, '.', count=1) if path.startswith(base) else path
```

Reply:
Ugly and more effort: whenever I see an except I have to look at the entire body of the try block and decide whether any of the code in there could be raising an exception that I did not intend to catch. This also means checking any functions that the code block is calling. I find it easier to reason about conditionals that avoid the exceptions in the first place.

Reply:
Updated (and some extra documentation for the function included too)